.github/workflows/changelog-cd.yaml

```diff
@@ -0,0 +1,135 @@
+
+      # Build and push the image
+      - name: Install docker buildx
+        uses: docker/setup-buildx-action@4fd812986e6c8c2a69e18311145f9371337f27d4 # v3.4.0
+
+      - name: Login to GitHub Container Registry
+        id: login-ghcr
+        uses: docker/login-action@0d4c9c5ea7693da7b068278f7b52bda2a190a446 # v3.2.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Prepare container image labels and tags
+        id: meta
+        uses: docker/metadata-action@8e5442c4ef9f78752691e2d8f8d19755c6f78e81 # v5.5.1
+        with:
+          images: |
+            ghcr.io/${{ github.repository_owner }}/changelog
+          flavor: |
+            latest=false
+          # Enable sha tag on branch push events and pull requests.
+          # Enable semver tags on tag push events, but don't overwrite major/minor tags for prereleases.
+          # Semver tags won't be generated except upon tag events.
+          tags: |
+            type=sha,prefix=v0.0.0-{{branch}}-,enable=${{ startsWith(github.ref, 'refs/heads/') }}
+            type=sha,prefix=v0.0.0-{{base_ref}}-,enable=${{ github.event_name == 'pull_request' }}
+            type=semver,pattern={{major}},value=${{ steps.setup.outputs.version }},enable=${{ steps.setup.outputs.is-prerelease != 'true' }}
+            type=semver,pattern={{major}}.{{minor}},value=${{ steps.setup.outputs.version }},enable=${{ steps.setup.outputs.is-prerelease != 'true' }}
+            type=semver,pattern={{version}},value=${{ steps.setup.outputs.version }}
+
+      - name: Build the container image and push
+        uses: docker/build-push-action@1ca370b3a9802c92e886402e0dd88098a2533b12 # v6.4.1
+        with:
+          context: ${{ env.TOOL_DIRECTORY }}
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          annotations: ${{ steps.meta.outputs.annotations }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          platforms: |
+            linux/amd64
+            linux/arm64
+          provenance: true
+          sbom: true
+
+      # File a new release with the tarballs attached
+      - name: Create a new GitHub release
+        if: ${{ steps.setup.outputs.should-release == 'true' }}
+        working-directory: ${{ env.TOOL_DIRECTORY }}
+        env:
+          VERSION: ${{ steps.setup.outputs.version }}
+          IS_PRERELEASE: ${{ steps.setup.outputs.is-prerelease }}
+        run: |
+          if [[ "${IS_PRERELEASE}" == 'true' ]]; then
+            EXTRA_FLAGS=("--prerelease")
+          else
+            EXTRA_FLAGS=("--latest")
+          fi
+
+          readarray -d '' RELEASE_TARBALLS < <(
+            find . -name '*.tar.gz' -print0
+          )
+
+          echo "Creating a release for ${VERSION} with files:"
+          for file in "${RELEASE_TARBALLS[@]}"; do
+            ls -lh "${file}"
+          done
```

Comment:
No need for a `for` loop here. You can just run `ls -lh "${RELEASE_TARBALLS[@]}"`

Reply:
Thanks - think I left that behind when I was working on the array building logic